File: server/src/Makc2021.Layer4.Domains.DummyMain/Queries/Item/Get/ItemGetQueryDomainOutput.cs
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2021.Layer3.Sample.Entities.DummyMain;
using Makc2021.Layer3.Sample.Entities.DummyMainDummyManyToMany;
using Makc2021.Layer3.Sample.Entities.DummyManyToMany;
using Makc2021.Layer3.Sample.Entities.DummyOneToMany;

namespace Makc2021.Layer4.Domains.DummyMain.Queries.Item.Get
{
    /// <summary>
    /// Вывод запроса на получение элемента в домене.
    /// </summary>
    public class ItemGetQueryDomainOutput
    {
        #region Properties

        /// <summary>
        /// Объект сущности "DummyMain".
        /// </summary>
        public DummyMainEntityObject ObjectOfDummyMainEntity { get; set; }

        /// <summary>
        /// Объекты сущности "DummyManyToMany".
        /// </summary>
        public DummyManyToManyEntityObject[] ObjectsOfDummyManyToManyEntity { get; set; }

        /// <summary>
        /// Объекты сущности "DummyMainDummyManyToMany".
        /// </summary>
        public DummyMainDummyManyToManyEntityObject[] ObjectsOfDummyMainDummyManyToManyEntity { get; set; }

        /// <summary>
        /// Объект сущности "DummyOneToMany".
        /// </summary>
        public DummyOneToManyEntityObject ObjectOfDummyOneToManyEntity { get; set; }

        #endregion Properties
    }
}

File: client/makc2021-react-redux-ts/src/Layer4/Domains/DummyMain/Queries/Item/Get/ItemGetQueryDomainOutput.ts
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

import { DummyMainEntityObject } from 'src/Layer3/Sample/Entities/DummyMainEntityObject';

/**
 * Выходные данные запроса на получение элемента в домене.
 */
export interface ItemGetQueryDomainOutput {
  /**
   * Объект сущности "DummyMain".
   */
  objectOfDummyMainEntity: DummyMainEntityObject;
}

/**
 * Создать выходные данные запроса на получение элемента в домене.
 * @returns Выходные данные запроса на получение элемента в домене.
 */
export function createItemGetQueryDomainOutput(): ItemGetQueryDomainOutput {
  return {} as ItemGetQueryDomainOutput;
}
